include/gatekeeper_gk.h

```diff
@@ -24,13 +24,113 @@
 #include "gatekeeper_ipip.h"
 #include "gatekeeper_ggu.h"
 #include "gatekeeper_mailbox.h"
+#include "gatekeeper_lpm.h"
+
+/* The maximum number of gateways that Gatekeeper may use. */
+#define GK_MAX_NUM_GW       (8)
+
+/*
+ * The LPM supports 8-bit next hop,
+ * which at most has 128 different values.
+ */
+#define GK_MAX_NUM_POLICIES (128)
 
 /*
  * A flow entry can be in one of three states:
  * request, granted, or declined.
  */
 enum gk_flow_state { GK_REQUEST, GK_GRANTED, GK_DECLINED };
 
+enum gk_policy_action {
+	/* Forward the packe to the corresponding gateway. */
+	GK_FWD_GATEWAY,
+
+	/*
+	 * The Grantor server is a neighbor.
+	 * Forward the packe to the Grantor server directly.
+	 */
+	GK_FWD_NEIGHBOR,
+
+	/* Forward the packet to the back interface. */
+	GK_FWD_BACK_NET,
+
+	/* Drop the packet. */
+	GK_DROP,
+
+	/* Invalid policy action. */
+	GK_POLICY_MAX,
+};
+
+struct gk_gateway {
```

Comment:
`gateway` is a bad name here because this struct may represent a gateway or a neighbor. Suggestion: `gk_nexthop`.